src/main/java/com/Novi/TechItEasy/services/RemoteControllerService.java

```diff
@@ -0,0 +1,126 @@
+package com.Novi.TechItEasy.services;
+
+import com.Novi.TechItEasy.dtos.RemoteControllerDto;
+import com.Novi.TechItEasy.dtos.RemoteControllerInputDto;
+import com.Novi.TechItEasy.exceptions.RecordNotFoundException;
+import com.Novi.TechItEasy.helpers.DtoConverters;
+import com.Novi.TechItEasy.models.RemoteController;
+import com.Novi.TechItEasy.repositories.RemoteControllerRepository;
+import org.springframework.stereotype.Service;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+
+@Service
+public class RemoteControllerService {
+
+    private final RemoteControllerRepository remoteControllerRepository;
+
+    public RemoteControllerService(RemoteControllerRepository remoteControllerRepository) {
+        this.remoteControllerRepository = remoteControllerRepository;
+    }
+
+
+    /// For fetching all remote controllers currently in the database /////
+    public List<RemoteControllerDto> getRemoteControllers() {
+
+        List<RemoteController> remoteControllers = new ArrayList<>(remoteControllerRepository.findAll());
+        List<RemoteControllerDto> remoteControllerDtos = new ArrayList<>();
+
+        for (RemoteController remoteController : remoteControllers) {
+            RemoteControllerDto dto = new RemoteControllerDto();
+            DtoConverters.remoteControllerDtoConverter(remoteController, dto);
+            remoteControllerDtos.add(dto);
+        }
+
+        if (remoteControllerDtos.isEmpty()) {
+            throw new RecordNotFoundException("We hebben geen afstandsbedieningen om te laten zien");
+        } else {
+            return remoteControllerDtos;
+        }
+    }
+
+
+    /// For fetching one remote controller by id /////
+    public RemoteControllerDto getRemoteController(Long id) {
+        Optional<RemoteController> fetchedRemoteController = remoteControllerRepository.findById(id);
+
+        if (fetchedRemoteController.isPresent()) {
+            RemoteControllerDto dto = new RemoteControllerDto();
+            DtoConverters.remoteControllerDtoConverter(fetchedRemoteController.get(), dto);
+            return dto;
+
+        } else {
+            throw new RecordNotFoundException("We hebben geen afstandsbediening met dit ID.");
```

Comment:
Misschien handig om in deze message ook aan te geven wat precies het ID is waas het om gaat.